### ISSUE TITLE
Please remove the Unicode version number from the C# language specification

### ISSUE THREAD

**Issue by gafter** [dotnet/csharplang#299]
The Roslyn C# compiler inherits its Unicode character classification from whatever is provided on the platform that hosts the compiler. Consequently we use a fairly recent Unicode version.

Please update the C# language specification to allow this behavior by removing the reference to a particular version of the Unicode specification.

(It is possible that this issue does not apply to a recent version of the spec, in which case you are welcome to close it)

**Comment by miloush** [comment]
While I am 100% for the latest Unicode support, I think the point of a specification is to provide definitive answers on what should work (and perhaps what should not).

Removing a reference to a particular version makes the whole Unicode support useless and the compilation results could vary wildly depending on the host.

How about specifying a minimum version to be supported?

**Comment by gafter** [comment]
The Unicode specification occasionally makes "breaking changes" such as changing a letter into a non-letter. Specifying a minimum version required would not protect against such breaks when they occur. Specifying a particular Unicode version would cause the compiler to have to include its own version-specific copy of the Unicode data tables. (Maybe that wouldn't be the worst thing in the world)

Did anyone notice that the Roslyn compilers don't accept letters composed of surrogate pairs? If not, why would they care much about which precise Unicode version is used?

**Comment by whoisj** [comment]
@gafter time to ship libiconv with every .NET Core executable? :wink:

**Comment by miloush** [comment]
@gafter I did. And I went to the specification hoping I could take it and wave it in front of people to defend my rights to proper Unicode handling. What I found was a reference to a Unicode version that does not define any characters beyond BMP, so I had no standing and stayed quiet.

If you remove the version anyone could come and say that's not a bug, the specification does not require us to support characters from extended planes.

My suggestion with minimum version was not meant to prevent breaking changes, but to establish minimum requirements (and perhaps scripts) that should be supported.
